tensorflow_addons/rnn/cell.py

```diff
@@ -799,3 +799,93 @@ def get_config(self):
         }
         base_config = super().get_config()
         return {**base_config, **config}
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class PeepholeLSTMCell(tf.keras.layers.LSTMCell):
+    """Equivalent to `tf.keras.layers.LSTMCell` class but adds peephole connections.
+
+    Peephole connections allow the gates to utilize the previous internal state as
+    well as the previous hidden state (which is what LSTMCell is limited to).
+    This allows PeepholeLSTMCell to better learn precise timings over LSTMCell.
+
+    From [Gers et al., 2002](
+    http://www.jmlr.org/papers/volume3/gers02a/gers02a.pdf):
+
+    "We find that LSTM augmented by 'peephole connections' from its internal
+    cells to its multiplicative gates can learn the fine distinction between
+    sequences of spikes spaced either 50 or 49 time steps apart without the help
+    of any short training exemplars."
+
+    The peephole implementation is based on:
+
+    [Sak et al., 2014](https://research.google.com/pubs/archive/43905.pdf)
+
+    Example:
+
+    ```python
+    # Create 2 PeepholeLSTMCells
+    peephole_lstm_cells = [PeepholeLSTMCell(size) for size in [128, 256]]
+    # Create a layer composed sequentially of the peephole LSTM cells.
+    layer = RNN(peephole_lstm_cells)
+    input = keras.Input((timesteps, input_dim))
+    output = layer(input)
+    ```
+    """
+
+    def build(self, input_shape):
+        super(PeepholeLSTMCell, self).build(input_shape)
```

Comment:
we could use `super().build(input_shape)` for consistent python3 style in the repo.

Reply:
Done.